pyaedt/icepak.py

```diff
@@ -6051,3 +6051,93 @@ def assign_conducting_plate_with_conductance(self, obj_plate, boundary_name=None
                                             shell_conduction=shell_conduction,
                                             low_side_rad_material=low_side_rad_material,
                                             high_side_rad_material=high_side_rad_material)
+
+    @pyaedt_function_handler()
+    def create_geometry_from_csv(self, csv_file, header_line=2, column_mapping=None, geo_type='cylinder', unit="mm"):
+        """Creates geometry from csv file containing geometry data.
+
+        Parameters
+        ----------
+        csv_file : str
+            Csv file containing the geometry details.
+            Expected column headers are dependent on the geometry type:
+
+            - if ``geo_type=='cylinder'``, expected fields are ``'xc'`` (base
+            center x coordinate),``'yc'`` (base center y coordinate), ``'zc'``
+            (base center z coordinate), ``'height'``, ``'radius'``,
+            ``'iradius'`` (inner radius for hollow cylinder), ``'plane'``
+            (plane on which the cylinder base lies).
+        header_line : int, optional
+            The line marking the start of data.
+            The default value is 2.
+        column_mapping : dict, optional
+            Map of the csv columns names to the accepted column names.
+        geo_type: str, optional
+            Type of geometry that the csv file describes. Accepted values are: ``"cylinder"``.
+            Default is 'cylinder'.
+        unit : str , optional
+            Unit of values used to create geometry. Default unit is "mm".
+
+        Returns
+        -------
+        list of :class:`pyaedt.modeler.cad.object3d.Object3d`
+        """
+        import pandas as pd
+        df = pd.read_csv(csv_file, header=header_line)
+        if column_mapping is not None:
+            df = df.rename(columns=column_mapping)
```

Comment:
You can test a csv with non-default column names. Just modify the current csv and add the argument`